File: credit.py
from cs50 import get_string

# Prompt for card number
n = get_string("Number:\n")

# Find length of card number
x = len(n)

# Find leading digits of card number
a = list(n)
y = int(a[0] + a[1])
z = int(a[0])

# Index of second to last card number
dec = 1
b = int(a[x - dec])

# Apply Luhn's Algorithm
temp = int(n)
sum_digits = 0

# Loop through every second digit from second to last digit
for i in range(b, 0, 1):
    # Check if digit is positive and greater than 9
    digit = temp % 10
    if i % 2 == 0:
        if digit * 2 > 9:
            sum_digits += (digit * 2) - 9
        else:
            sum_digits += digit * 2
    else:
        sum_digits += digit
    temp /= 10

    dec += 2


# Declare bool to test algorithm result
luhn = None
if sum_digits % 10 == 0:
    luhn = True

# Check card numbers against conditions
if y == 34 or y == 37 and x == 15 and luhn:
    print("AMEX\n")

elif y > 50 and y < 56 and x == 16 and luhn:
    print("MASTERCARD\n")

elif (x == 13 or x == 16) and (z == 4 and luhn):
    print("VISA\n")

else:
    print("INVALID\n")

File: PSET 1/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long n;
    do
    {
        n = get_long("Number: ");
    }
    while (n < 1 || n > 9999999999999999);

    long temp_n = n;
    int c = 2;

    while (temp_n > 100)
    {
        temp_n = temp_n / 10;
        c++;
    }

    long temp_n2 = n;
    int sum = 0;

    for (int i = 1; i <= c; i++)
    {
        int digit = temp_n2 % 10;
        if (i % 2 == 0)
        {
            if (digit * 2 > 9)
            {
                sum += (digit * 2) - 9;
            }
            else
            {
                sum += digit * 2;
            }
        }
        else
        {
            sum += digit;
        }  
        temp_n2 /= 10;
    }

    bool flag = (sum % 10 == 0) ? true : false;
  
    // Combine all parameters to check card numbers
    if (temp_n > 50 && temp_n < 56 && c == 16 && flag)
    {
        printf("MASTERCARD\n");
    }
    else if ((temp_n == 34 || temp_n == 37) && (c == 15) && flag)
    {
        printf("AMEX\n");
    }
    else if ((c == 13 || c == 16) && (temp_n / 10 == 4) && flag)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
